src/app/screens/signBatchPsbtRequest/index.tsx

```diff
@@ -0,0 +1,763 @@
+  //   keyPrefix: 'SIGNATURE_REQUEST',
+  // });
+  const [expandInputOutputView, setExpandInputOutputView] = useState(false);
+  const {
+    payload,
+    confirmSignPsbt,
+    broadcastPsbt,
+    cancelSignPsbt,
+    getSigningAddresses,
+    requestToken,
+  } = useSignBatchPsbtTx();
+  const [isSigning, setIsSigning] = useState(false);
+  const [isBroadcasting, setIsBroadcasting] = useState(false);
+  const [isBroadcastingComplete, setIsBroadcastingComplete] = useState(false);
+  const [isSigningComplete, setIsSigningComplete] = useState(false);
+  const [signingPsbtIndex, setSigningPsbtIndex] = useState(0);
+  const [broadcastingPsbtIndex, setBroadcastingPsbtIndex] = useState(0);
+  const [hasOutputScript, setHasOutputScript] = useState(false);
+  const [currentPsbtIndex, setCurrentPsbtIndex] = useState(0);
+  const [reviewTransaction, setReviewTransaction] = useState(false);
+  // const [isModalVisible, setIsModalVisible] = useState(false);
+  // const [currentStepIndex, setCurrentStepIndex] = useState(0);
+  // const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+  // const [isConnectSuccess, setIsConnectSuccess] = useState(false);
+  // const [isConnectFailed, setIsConnectFailed] = useState(false);
+  // const [isTxApproved, setIsTxApproved] = useState(false);
+  // const [isTxRejected, setIsTxRejected] = useState(false);
+  const { search } = useLocation();
+  const params = new URLSearchParams(search);
+  const tabId = params.get('tabId') ?? '0';
+  // const request = decodeToken(requestToken) as any as SignTransactionOptions;
+  // const btcClient = useBtcClient();
+
+  const handlePsbtParsing = useCallback(
+    (item) => {
+      try {
+        return parsePsbt(selectedAccount!, item.inputsToSign, item.psbtBase64, network.type);
+      } catch (err) {
+        return '';
+      }
+    },
+    [selectedAccount, network.type],
+  );
+
+  const parsedPsbts = useMemo(
+    () => payload.psbts.map(handlePsbtParsing),
+    [handlePsbtParsing, payload.psbts],
+  );
+
+  const userReceivesOrdinalArr = useDetectOrdinalInSignBatchPsbt(parsedPsbts);
+
+  const checkIfMismatch = () => {
+    if (!parsedPsbts?.length) {
+      navigate('/tx-status', {
+        state: {
+          txid: '',
+          currency: 'BTC',
+          errorTitle: t('PSBT_CANT_PARSE_ERROR_TITLE'),
+          error: t('PSBT_CANT_PARSE_ERROR_DESCRIPTION'),
+          browserTx: true,
+        },
+      });
+    }
+    if (payload.network.type !== network.type) {
+      navigate('/tx-status', {
+        state: {
+          txid: '',
+          currency: 'BTC',
+          error: t('NETWORK_MISMATCH'),
+          browserTx: true,
+        },
+      });
+    }
+
+    const checkAddressMismatch = (input) => {
+      if (input.address !== btcAddress && input.address !== ordinalsAddress) {
+        navigate('/tx-status', {
+          state: {
+            txid: '',
+            currency: 'STX',
```

Comment:
wrong currency here

Reply:
I copied this logic from `src/app/screens/signPsbtRequest/index.tsx`, I'll update it in both places